From dc0fed370865b49e9bcb2db2eb7ff21adec904a8 Mon Sep 17 00:00:00 2001
From: Jim Ramsay <lack@gentoo.org>
Date: Tue, 24 Aug 2010 12:23:25 -0400
Subject: [PATCH] Move GrabWeather script to $(PREFIX)/libexec/gkrellweather/

This is to avoid a file collision with x11-misc/bbweather (Bug #333289)
---
 Makefile        |    4 ++--
 gkrellweather.c |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index d6dc404..693b4f2 100644
--- a/Makefile
+++ b/Makefile
@@ -38,13 +38,13 @@ install:
 	else \
 		install -D -c -s -m 644 gkrellweather.so $(PREFIX)/lib/gkrellm2/plugins/gkrellweather.so ; \
 	fi
-	install -c -m 755 GrabWeather $(PREFIX)/bin
+	install -D -c -m 755 GrabWeather $(PREFIX)/libexec/gkrellweather/
 
 uninstall:
 	(cd po && ${MAKE} uninstall )
 	rm -f /usr/lib/gkrellm2/plugins/gkrellweather.so
 	rm -f $(PREFIX)/lib/gkrellm2/plugins/gkrellweather.so
-	rm -f $(PREFIX)/bin/GrabWeather
+	rm -f $(PREFIX)/libexec/gkrellweather/GrabWeather
 
 dist:
 	rm -rf $(PKGNAME)-$(VERSION)
diff --git a/gkrellweather.c b/gkrellweather.c
index 1d4cf4e..224260a 100644
--- a/gkrellweather.c
+++ b/gkrellweather.c
@@ -721,7 +721,7 @@ apply_air_config (void)
         options.station[1] = c[1];
         options.station[2] = c[2];
         options.station[3] = c[3];
-        snprintf(options.command, 512, PREFIX "/bin/GrabWeather %s", options.station);
+        snprintf(options.command, 512, PREFIX "/libexec/gkrellweather/GrabWeather %s", options.station);
         options.command[511] = 0;
         snprintf(options.filename, 512, "%s/.wmWeatherReports/%s.dat",
                 getenv("HOME"), options.station);
@@ -986,7 +986,7 @@ read_default(void)
     options.windspeed_unit = 1;
     options.pressure_unit = 1;
     strcpy(options.station, DEFAULT_STATION_ID);
-    snprintf(options.command, 512, "/usr/share/gkrellm/GrabWeather %s", options.station);
+    snprintf(options.command, 512, PREFIX "/libexec/gkrellweather/GrabWeather %s", options.station);
     options.command[511] = 0;
     snprintf(options.filename, 512, "%s/.wmWeatherReports/%s.dat",
              getenv("HOME"), options.station);
-- 
1.7.2

